cmd/cosign/cli/sign/sign.go

```diff
@@ -431,14 +382,11 @@ func keylessSigner(ctx context.Context, ko KeyOpts) (*SignerVerifier, error) {
 
 	if ko.InsecureSkipFulcioVerify {
 		k, err = fulcio.NewSigner(ctx, tok, ko.OIDCIssuer, ko.OIDCClientID, fClient)
-		if err != nil {
-			return nil, errors.Wrap(err, "getting key from Fulcio")
-		}
 	} else {
 		k, err = fulcioverifier.NewSigner(ctx, tok, ko.OIDCIssuer, ko.OIDCClientID, fClient)
-		if err != nil {
-			return nil, errors.Wrap(err, "getting key from Fulcio")
-		}
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "getting key from Fulcio")
```

Comment:
This saves ~3 LoC and is less clear, I'd revert.

If you want to save 2 LoC without losing clarity, you can fold the call into the `if`:
```go
if k, err = fulcio.NewSigner(...); err != nil {
     return nil, errors.Wrap(err, "getting key from Fulcio")
}
```

Reply:
Also, I would expect that ultimately the bonus logic in `fulcioverifier` just becomes yet-another composition of the `Signer` interface.

Reply:
right, already have a TODO for it